.github/workflows/ubuntu-win-generation.yml

```diff
@@ -0,0 +1,123 @@
+name: MMS image generation
+on:
+  workflow_call:
+    inputs:
+      image_name:
+        type: string
+        description: An OS image to build
+        required: true
+      image_readme_name:
+        type: string
+        description: README file path
+        required: true
+      custom_repo:
+        type: string
+        description: Custom repo to checkout
+        required: false
+      custom_repo_hash:
+        type: string
+        description: Custom repo hash
+        required: false
+defaults:
+  run:
+    shell: pwsh
+
+jobs:
+  build:
+    runs-on: ci-agent
+    timeout-minutes: 1200
+    steps:
+      - name: Checkout repository
+        if: ${{ (inputs.custom_repo == '' && inputs.custom_repo_hash == '') || github.event_name != 'pull_request_target'  }}
+        uses: actions/checkout@v3
+        with:
+          repository: actions/runner-images
+
+      - name: Checkout PR
+        if: ${{ github.event_name == 'pull_request_target' && github.event.action == 'labeled' }}
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Checkout custom repository
+        if: ${{ inputs.custom_repo != '' && inputs.custom_repo_hash != ''  }}
+        uses: actions/checkout@v3
+        with:
+          repository: '${{ inputs.custom_repo }}'
+          ref: '${{ inputs.custom_repo_hash }}'
+
+      - name: Set image varibles
+        run: |
+          $ImageType = "${{ inputs.image_name }}"
+
+          if ($ImageType.StartsWith("ubuntu")) { $TemplateDirectoryName = "linux" } else { $TemplateDirectoryName = "win" }
+
+          $TemplateDirectoryPath = Join-Path "images" $TemplateDirectoryName | Resolve-Path
+          $TemplatePath = Join-Path $TemplateDirectoryPath "$ImageType.pkr.hcl"
+
+          if ( -not (Test-Path $TemplatePath) ) {
+            $TemplatePath = Join-Path $TemplateDirectoryPath "$ImageType.json"
+          }
+
+          echo "::set-env name=TemplatePath::$TemplatePath"
+          echo "::set-env name=TemplateDirectoryPath::$TemplateDirectoryPath"
+          echo "::set-env name=ImageType::$ImageType"
+        env:
+           ACTIONS_ALLOW_UNSECURE_COMMANDS: true
```

Comment:
Any reason why we can't use secure commands to set environment variables?

Reply:
Do you mean using something like `"VAR=value" >> $env:GITHUB_ENV` ?

Reply:
Yes, exactly.
```pwsh
"VAR=value" | Out-File -Append -FilePath $env:GITHUB_ENV
```

I think this old unsecure way will be deprecated soon or later so it makes sense to switch to the new approach.

Reply:
Updated